### PR TITLE
Remove numbering, set list of files

### LINKED ISSUE #1023: Remove clause/section numbering
There should be no numbering on chapter and section headers.

It is automatically added while creating the PDF.

### LINKED ISSUE #1040: [3.0.1] Annexes have to be renumber + Removing Annex files that no longer in use
- Heading numbers will be automatically numbered
- Some Annexes are now in spdx/using repo, have to remove those files

### LINKED ISSUE #1041: [3.0.1] Consistency of Headings
Current we have two styles of headings, including ones in the navigation bar

- Heading like this; and
- Heading Like This

Have to make them consistent for the publication

(The "Diffs" in "Diffs from Previous Version", may have to expand to full word as well)

Screenshot 2024-08-10 at 18 09 56

Screenshot 2024-08-10 at 18 11 43

Screenshot 2024-08-10 at 18 11 32

Screenshot 2024-08-10 at 18 11 19

Screenshot 2024-08-10 at 18 15 35

### PR DESCRIPTION
**Pull request by zvr**
- Removes numbering from chapters/sections in main spec
- Does the same for annexes (removing unused files)
- Updates mkdocs configuration with chapters and annexes

Also renames some files, changes titles to consistent casing, etc.

Closes #1023
Closes #1040
Closes #1041

### CODE REVIEW

File: docs/annexes/SPDX-Lite.md [hunk 1]
```diff
@@ -1,18 +1,17 @@
-# Annex H: SPDX Lite  
+# SPDX Lite  
 
-## H.1 Definition of the Lite profile <a name="H.1"></a>  
+## Definition of the Lite profile
```
Comment (bact) [inline docs/annexes/SPDX-Lite.md:3]:
```suggestion
## Definition of the Lite profile <a name="H.1"></a>
```

Add back the HTML anchor, as they are also appeared in other files.
Please kindly correct the "H", if it is no longer the right numbering.

File: docs/annexes/SPDX-Lite.md [hunk 2]
```diff
@@ -1,18 +1,17 @@
...
 
 The Lite profile is designed to make it quick and easy to start a Software Bill of Materials in situations where a company may have limited capacity for introducing new items into its process.  
 The Lite profile captures the minimum set of information required for license compliance in the software supply chain. It contains information about the creation of the SBOM, package lists with licensing and other related items, and their relationships.  
 
 All elements in Lite profile are essential for complying with licenses. It is easy to use a SPDX document with the Lite profile for anyone who does not have enough knowledge about licensing information and easy to import license information from former versions of SPDX Lite format files.  
 The Lite profile offers the flexibility to be used either alone or in combination with other SPDX profiles as a SPDX document in the software supply chain.  
 
-## H.2 Table of the Lite profile elements <a name="H.2"></a>  
+## Table of the Lite profile elements
```
Comment (bact) [inline docs/annexes/SPDX-Lite.md:11]:
```suggestion
## Table of the Lite profile elements <a name="H.2"></a>
```

**Review by bact — CHANGES_REQUESTED** [review]
Add back HTML anchors in SPDX Lite annex, as they are also appeared in other files.

File: docs/annexes/SPDX-Lite.md (hunk 1 above)
Comment (zvr) [reply docs/annexes/SPDX-Lite.md:3]: We should point to the Annex as a file, not subsections.

File: docs/annexes/SPDX-Lite.md (hunk 2 above)
Comment (zvr) [reply docs/annexes/SPDX-Lite.md:11]: After I finish fixing #1043, there will not be a section

File: docs/annexes/SPDX-Lite.md (hunk 1 above)
Comment (bact) [reply docs/annexes/SPDX-Lite.md:3]: If so, we should remove the HTML anchors in other files as well.

File: docs/annexes/SPDX-license-expressions.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
-# Annex D: SPDX license expressions (Normative)
+# SPDX license expressions (Normative)
 
-## D.1 Overview <a name="D.1"></a>
+## Overview <a name="D.1"></a>
```
Comment (bact) [inline docs/annexes/SPDX-license-expressions.md:3]:
```suggestion
## Overview
```
If subsection reference is not desirable, removing the HTML anchor.

Comment (bact) [reply docs/annexes/SPDX-license-expressions.md:3]: For cleaner URLs, propose to rename these files:
- `SPDX-Lite.md` to `spdx-lite.md`
- `SPDX-license-expressions.md` to `spdx-license-expressions.md`

Other files are already in all-lowercase, including ones with a word which originally an all-uppercase initialism:
- `changes-from-previous-iso.md` (ISO)
- `pkg-url-specification.md` (URL)
- `rdf-model.md` (RDF)

The all-lowercase `spdx` will also match the other `spdx` instances in the full URL of: `https://spdx.github.io/spdx-spec/...`

File: docs/annexes/SPDX-Lite.md (hunk 1 above)
Comment (zvr) [reply docs/annexes/SPDX-Lite.md:3]: If we wait to do everything in this PR, it will never get merged...

Comment (bact) [reply docs/annexes/SPDX-Lite.md:3]: No prob at all, can do those in separated PRs too.

File: docs/annexes/SPDX-license-expressions.md (hunk 3 above)
Comment (bact) [reply docs/annexes/SPDX-license-expressions.md:3]: I will put this in another PR.

File: docs/annexes/SPDX-Lite.md (hunk 2 above)
Comment (kestewart) [reply docs/annexes/SPDX-Lite.md:11]: Discussed in meeting, and we'll see what's resolved with auto generation, and we'll fix later.

**Review by kestewart — APPROVED** [review]
Discussed this in call about anchors,  and don't think this will impact pdf, so going ahead and merging.

**Review by goneall — APPROVED** [review]
LGTM

**Review by bobmartin3000 — APPROVED** [review]
LGTM

**Review by bact — APPROVED** [review]
Looks good - Will do the filenames later

**Pull request merged**